src/app/modules/submission/email-submission/email-submission.middleware.ts

```diff
@@ -28,3 +41,107 @@ export const prepareEmailSubmission: RequestHandler<
+ * req.body.
+ *
+ * @param req - Express request object
+ * @param res - Express response object
+ * @param next - Express next middleware function
+ */
+export const receiveEmailSubmission: RequestHandler<
+  ParamsDictionary,
+  { message: string },
+  { responses: FieldResponse[] }
+> = async (req, res, next) => {
+  const logMeta = {
+    action: 'receiveEmailSubmission',
+    formId: (req as WithForm<typeof req>).form._id,
+    ...createReqMeta(req),
+  }
+  return EmailSubmissionReceiver.createMultipartReceiver(req.headers)
+    .asyncAndThen((receiver) => {
+      const result = EmailSubmissionReceiver.configureMultipartReceiver(
+        receiver,
+      )
+      req.pipe(receiver)
+      return result
+    })
+    .map((parsed) => {
+      req.body = parsed
+      const hashes = EmailSubmissionService.hashSubmission(parsed)
+      logger.info({
+        message: 'Submission successfully hashed',
+        meta: { ...logMeta, ...hashes },
+      })
+      return next()
+    })
+    .mapErr((error) => {
+      logger.error({
+        message: 'Error while receiving multipart data',
+        meta: logMeta,
+        error,
+      })
+      const { errorMessage, statusCode } = mapRouteError(error)
+      return res.status(statusCode).json({ message: errorMessage })
+    })
+}
+
+/**
+ * Extracts relevant fields, injects questions, verifies visibility of field and validates answers
+ * to produce req.body.parsedResponses
+ *
+ * @param req - Express request object
+ * @param res - Express response object
+ * @param next - Express next middleware function
+ */
+export const validateEmailSubmission: RequestHandler<
+  ParamsDictionary,
+  { message: string },
+  { responses?: FieldResponse[]; parsedResponses: ProcessedFieldResponse[] }
+> = async (req, res, next) => {
+  const { form } = req as WithForm<typeof req>
+
+  if (!req.body.responses) {
+    return res.sendStatus(StatusCodes.BAD_REQUEST)
+  }
+
+  return EmailSubmissionService.validateAttachments(req.body.responses)
+    .andThen(() => getProcessedResponses(form, req.body.responses!))
+    .map((parsedResponses) => {
+      // Creates an array of attachments from the validated responses
+      // eslint-disable-next-line @typescript-eslint/no-extra-semi
+      ;(req as WithAttachments<
+        typeof req
+      >).attachments = mapAttachmentsFromResponses(req.body.responses!)
+      req.body.parsedResponses = parsedResponses
+      delete req.body.responses // Prevent downstream functions from using responses by deleting it
+      return next()
+    })
+    .mapErr((error) => {
+      logger.error({
+        message:
+          error instanceof ConflictError
+            ? 'Conflict - Form has been updated'
+            : 'Error processing responses',
+        meta: {
+          action: 'validateEmailSubmission',
+          ...createReqMeta(req),
+          formId: form._id,
+        },
+        error,
+      })
+      if (error instanceof ConflictError) {
```

Comment:
should we still use mapRouteError util for this?

Reply:
yeah good point. can probably clean this part up. previously I just copied it over.